app/templates/client/app/main/main.controller(js).js

```diff
@@ -1,27 +1,30 @@
 'use strict';
 
-angular.module('<%= scriptAppName %>')
-  .controller('MainCtrl', function($scope, $http<% if (filters.socketio) { %>, socket<% } %>) {
-    $scope.awesomeThings = [];
+function MainController($scope, $http<% if (filters.socketio) { %>, socket<% } %>) {
+  var self = this;
+  this.awesomeThings = [];
 
-    $http.get('/api/things').then(function(response) {
-      $scope.awesomeThings = response.data;<% if (filters.socketio) { %>
-      socket.syncUpdates('thing', $scope.awesomeThings);<% } %>
-    });
+  $http.get('/api/things').then(function(response) {
+    self.awesomeThings = response.data;<% if (filters.socketio) { %>
+    socket.syncUpdates('thing', self.awesomeThings);<% } %>
+  });
 <% if (filters.models) { %>
-    $scope.addThing = function() {
-      if ($scope.newThing === '') {
-        return;
-      }
-      $http.post('/api/things', { name: $scope.newThing });
-      $scope.newThing = '';
-    };
+  this.addThing = function() {
+    if (self.newThing === '') {
+      return;
+    }
+    $http.post('/api/things', { name: self.newThing });
+    self.newThing = '';
+  };
 
-    $scope.deleteThing = function(thing) {
-      $http.delete('/api/things/' + thing._id);
-    };<% } %><% if (filters.socketio) { %>
+  this.deleteThing = function(thing) {
+    $http.delete('/api/things/' + thing._id);
+  };<% } %><% if (filters.socketio) { %>
 
-    $scope.$on('$destroy', function() {
-      socket.unsyncUpdates('thing');
-    });<% } %>
-  });
+  $scope.$on('$destroy', function() {
+    socket.unsyncUpdates('thing');
+  });<% } %>
+}
+
+angular.module('<%= scriptAppName %>')
+  .controller('MainController', MainController);
```

Comment:
I put the function before, since we'll want to use classes with es2015 code, and classes aren't hoisted. Consistency.